VehicleLaunchSFColumn/js/bar.js

```diff
@@ -0,0 +1,189 @@
+Highcharts.chart("hcContainer", {
+  data: {
+    googleSpreadsheetKey: "11gJ3QuXf4_34uDomERHdO3FQNLZ2_6_t8dY_SZKaESo",
+    googleSpreadsheetWorksheet: 1,
+  },
+  chart: {
+    type: "column",
+  },
+
+  title: {
+    text: "Rocket Successes and Failures",
+  },
+
+  xAxis: {
+    categories: [
+      "2002",
+      "2003",
+      "2004",
+      "2005",
+      "2006",
+      "2007",
+      "2008",
+      "2009",
+      "2010",
+      "2011",
+      "2012",
+      "2013",
+      "2014",
+      "2015",
+      "2016",
+      "2017",
+      "2018",
+      "2019",
+      "2020",
+    ],
+  },
+
+  yAxis: {
+    allowDecimals: false,
+    min: 0,
+    title: {
+      text: "Number of Launches",
+    },
+    reversedStacks: false,
+  },
+
+  credits: {
+    enabled: true,
+    href: false,
+    text:
+      "CSIS Aerospace Security | Gunter's Space Page, SpaceX, United Launch Alliance",
+  },
+
+  plotOptions: {
+    column: {
+      stacking: "normal",
+    },
+  },
+
+  tooltip: {
+    shared: true,
+    useHTML: true,
+    formatter: function () {
+      let string = "";
+
+      for (i = 0; i < this.points.length; i += 2) {
+        string +=
+          "<tr>" +
+          "<td><b>" +
+          this.points[i].series.userOptions.stack +
+          "</b></td>" +
+          "<td>" +
+          this.points[i].y +
+          "</td>" +
+          "<td>" +
+          this.points[i + 1].y +
+          "</td>" +
+          "</tr>";
+      }
+
+      return (
+        '<span style="font-size: 14px;"><b>' +
+        this.x +
+        "</b></span>" +
+        "<br>" +
+        "<table>" +
+        "<tr>" +
+        "<th></th>" +
+        "<th> Success </th>" +
+        "<th> Failure </th>" +
+        "</tr>" +
+        string +
+        "</table>"
+      );
+    },
+  },
+
+  legend: {
+    labelFormatter: function () {
+      if (
+        this.name === "Delta IV Success" ||
+        this.name === "Atlas V Success" ||
+        this.name === "Falcon 9 Success"
+      ) {
+        let name = this.name.split("");
+        let updateName = name.slice(0, 8);
+        finalName = updateName.join("");
+        return finalName;
+      } else if (
+        this.name === "Delta IV Heavy Success" ||
+        this.name === "Falcon 9 Heavy Success"
+      ) {
+        let name = this.name.split("");
+        let updateName = name.slice(0, 14);
+        finalName = updateName.join("");
+        return finalName;
+      } else if (this.name === "Delta IV Failure/Partial Failure") {
+        let name = this.name.split("");
+        let updateName = name.slice(9, 13);
+        finalName = updateName.join("") + "ed Launches";
+        return finalName;
+      }
+    },
+  },
```

Comment:
This can be refactored by doing something like this
```suggestion
    legend: {
    labelFormatter: function () {
      if (this.name === "Delta IV Failure/Partial Failure") {
        let name = this.name.split("");
        let updateName = name.slice(9, 13);
        finalName = updateName.join("") + "ed Launches";
        return finalName;
      } else {
        return this.name.replace(/success/gi, "");
      }
    },
  },

Reply:
AMAZING.